src/components/RedeemGift/RedeemGift.tsx

```diff
@@ -0,0 +1,178 @@
+import { useAppDispatch, useAppSelector } from 'redux/store';
+import { fetchRedeemGift, fetchVerifyGift, selectGift } from 'redux/giftSlice';
+import { fetchOffer, selectOffer } from 'redux/offerSlice';
+import { ReactComponent as CheckmarkIcon } from 'assets/images/greenCheckmark.svg';
+import { getData } from 'util/appConfigHelper';
+import { LinkStyled } from 'components/ThankYouPage/ThankYouPageStyled';
+import Header from 'components/Header';
+import Footer from 'components/Footer';
+import Button from 'components/Button';
+import SectionHeader from 'components/SectionHeader';
+import MyAccountInput from 'components/MyAccountInput';
+import OfferCheckoutCard from 'components/OfferCheckoutCard';
+import Loader from 'components/Loader';
+import {
+  HeaderStyled,
+  InfoTextStyled,
+  InputWrapperStyled,
+  OfferWrapperStyled,
+  RedeemGiftWrapperStyled,
+  ThankYouPageStyled,
+  WrapperStyled
+} from './RedeemGift.styled';
+
+type RedeemGiftProps = {
+  onBackClick: () => void;
+  onSuccess: (...args: unknown[]) => void;
+};
+
+const RedeemGift = ({ onBackClick, onSuccess }: RedeemGiftProps) => {
+  const [giftCode, setGiftCode] = useState('');
+  const [showOffer, setShowOffer] = useState(false);
+  const [isGiftRedeemed, setIsGiftRedeemed] = useState(false);
+
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+
+  const {
+    error,
+    isRedeemLoading,
+    isVerifyLoading,
+    verifiedGift: { redeemable }
+  } = useAppSelector(selectGift);
+
+  const { loading: isOfferLoading } = useAppSelector(selectOffer);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setGiftCode(e.target.value);
+
+  const handleVerifyGift = async () => {
+    const resultVerifyGiftAction = await dispatch(fetchVerifyGift(giftCode))
+      .unwrap()
+      .catch(err => {
+        setShowOffer(false);
+        throw new Error(err);
+      });
+
+    const { offerId } = resultVerifyGiftAction;
+
+    if (offerId) {
+      setShowOffer(true);
+      await dispatch(fetchOffer(offerId));
+    }
+  };
+
+  const handleRedeemGift = async () => {
+    await dispatch(fetchRedeemGift(giftCode))
+      .unwrap()
+      .catch(err => {
+        throw new Error(err);
+      });
+
+    setIsGiftRedeemed(true);
+  };
+
+  useEffect(() => {
+    const urlParams = new URLSearchParams(window.location.search);
+    const giftCodeParam = urlParams.get('giftCode');
+
+    if (giftCodeParam) {
+      setGiftCode(giftCodeParam);
+    }
+  }, [window.location.search]);
+
+  useEffect(() => {
+    let timer: ReturnType<typeof setTimeout>;
+
+    if (isGiftRedeemed) {
+      timer = setTimeout(() => {
+        onSuccess();
+      }, 3000);
+    }
+    return () => clearTimeout(timer);
+  }, [isGiftRedeemed]);
+
+  const isConfirmButtonDisabled =
+    !redeemable || isVerifyLoading || isOfferLoading;
+
+  return (
+    <WrapperStyled>
+      <Header onBackClick={onBackClick} />
+      {isGiftRedeemed ? (
```

Comment:
I'd consider break this jsx into two separate component. One for isGiftRedeemed === true and another when it is false. It should improve readability.